pkg/pmem-csi-operator/controller/deployment/deployment_controller_test.go

```diff
@@ -547,6 +548,62 @@ func TestDeploymentController(t *testing.T) {
 				})
 			}
 		})
+
+		t.Run("delete obsolete objects", func(t *testing.T) {
```

Comment:
We also need a negative test: "not delete unrelated objects".

Reply:
> We also need a negative test: "not delete unrelated objects".

Extended the test to this.